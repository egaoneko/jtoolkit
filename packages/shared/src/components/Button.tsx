import { FC, useState } from 'react';

const Button: FC = () => {
  const [count, setCount] = useState(0);

  return (
    <button>
      <button onClick={() => setCount(count + 1)}>Click me</button>
    </button>
  );
};

export default Button;
