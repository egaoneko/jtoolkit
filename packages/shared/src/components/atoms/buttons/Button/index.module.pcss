.button {
  @apply relative inline-flex items-center justify-center py-2.5 px-5 overflow-hidden;
  @apply rounded-md hover:shadow-lg focus:outline-none;
  @apply disabled:opacity-30 disabled:cursor-not-allowed;

  &.small {
    @apply text-sm;
  }

  &.medium {
    @apply text-base;
  }

  &.large {
    @apply text-lg;
  }

  &.rounded {
    @apply rounded-full;
  }

  &:not(:first-child) {
    @apply ml-2;
  }

  &.group {
    @apply ml-0 rounded-none;

    &:first-child {
      &.row {
        @apply rounded-l-md;
      }
      &.column {
        @apply rounded-t-md;
      }
    }

    &:last-child {
      &.row {
        @apply rounded-r-md;
      }
      &.column {
        @apply rounded-b-md;
      }
    }
  }

  & span.ripple {
    @apply absolute w-full h-full;

    & .effect {
      @apply absolute rounded-[50%] bg-white animate-ripple;
    }
  }
}

.contained {
  @apply text-white;

  &.primary {
    @apply bg-bgPrimaryDark hover:bg-bgPrimaryDarker;
    @apply dark:bg-dark-bgPrimaryDark hover:dark:bg-dark-bgPrimaryDarker;
  }

  &.secondary {
    @apply bg-bgSecondaryDark hover:bg-bgSecondaryDarker;
    @apply dark:bg-dark-bgSecondaryDark hover:dark:bg-dark-bgSecondaryDarker;
  }

  &.success {
    @apply bg-bgSuccessDark hover:bg-bgSuccessDarker;
    @apply dark:bg-dark-bgSuccessDark hover:dark:bg-dark-bgSuccessDarker;
  }

  &.danger {
    @apply bg-bgDangerDark hover:bg-bgDangerDarker;
    @apply dark:bg-dark-bgDangerDark hover:dark:bg-dark-bgDangerDarker;
  }

  &.warning {
    @apply bg-bgWarningDark hover:bg-bgWarningDarker;
    @apply dark:bg-dark-bgWarningDark hover:dark:bg-dark-bgWarningDarker;
  }

  &.info {
    @apply bg-bgInfoDark hover:bg-bgInfoDarker;
    @apply dark:bg-dark-bgInfoDark hover:dark:bg-dark-bgInfoDarker;
  }

  &.relief {
    @apply border-b-4;

    &.primary {
      @apply border-bgPrimaryDarker hover:bg-bgPrimary;
      @apply dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimary;
    }

    &.secondary {
      @apply border-bgSecondaryDarker hover:bg-bgSecondary;
      @apply dark:border-dark-bgSecondaryDarker hover:bg-bgSecondary;
    }

    &.success {
      @apply border-bgSuccessDarker hover:bg-bgSuccess;
      @apply dark:border-dark-bgSuccessDarker hover:bg-bgSuccess;
    }

    &.danger {
      @apply border-bgDangerDarker hover:bg-bgDanger;
      @apply dark:border-dark-bgDangerDarker hover:bg-bgDanger;
    }

    &.warning {
      @apply border-bgWarningDarker hover:bg-bgWarning;
      @apply dark:border-dark-bgWarningDarker hover:bg-bgWarning;
    }

    &.info {
      @apply border-bgInfoDarker hover:bg-bgInfo;
      @apply dark:border-dark-bgInfoDarker hover:bg-bgInfo;
    }
  }

  &.gradient {
    @apply bg-gradient-to-r;

    &.primary {
      @apply from-bgPrimaryLight to-bgPrimaryDarker;
      @apply dark:from-bgPrimaryLight dark:to-bgPrimaryDarker;
    }

    &.secondary {
      @apply from-bgSecondaryLight to-bgSecondaryDarker;
      @apply dark:from-bgSecondaryLight dark:to-bgSecondaryDarker;
    }

    &.success {
      @apply from-bgSuccessLight to-bgSuccessDarker;
      @apply dark:from-bgSuccessLight dark:to-bgSuccessDarker;
    }

    &.danger {
      @apply from-bgDangerLight to-bgDangerDarker;
      @apply dark:from-bgDangerLight dark:to-bgDangerDarker;
    }

    &.warning {
      @apply from-bgWarningLight to-bgWarningDarker;
      @apply dark:from-bgWarningLight dark:to-bgWarningDarker;
    }

    &.info {
      @apply from-bgInfoLight to-bgInfoDarker;
      @apply dark:from-bgInfoLight dark:to-bgInfoDarker;
    }
  }
}

.outlined {
  @apply border;

  &.primary {
    @apply text-textPrimaryDarker border-bgPrimaryDarker hover:bg-bgPrimaryLighter/[.05];
    @apply dark:text-dark-textPrimaryDarker dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimaryLighter/[.05];
  }

  &.secondary {
    @apply text-textSecondaryDarker border-bgSecondaryDarker hover:bg-bgSecondaryLighter/[.05];
    @apply dark:text-dark-textSecondaryDarker dark:border-dark-bgSecondaryDarker hover:dark:bg-dark-bgSecondaryLighter/[.05];
  }

  &.success {
    @apply text-textSuccessDarker border-bgSuccessDarker hover:bg-bgSuccessLighter/[.05];
    @apply dark:text-dark-textSuccessDarker dark:border-dark-bgSuccessDarker hover:dark:bg-dark-bgSuccessLighter/[.05];
  }

  &.danger {
    @apply text-textDangerDarker border-bgDangerDarker hover:bg-bgDangerLighter/[.05];
    @apply dark:text-dark-textDangerDarker dark:border-dark-bgDangerDarker hover:dark:bg-dark-bgDangerLighter/[.05];
  }

  &.warning {
    @apply text-textWarningDarker border-bgWarningDarker hover:bg-bgWarningLighter/[.05];
    @apply dark:text-dark-textWarningDarker dark:border-dark-bgWarningDarker hover:dark:bg-dark-bgWarningLighter/[.05];
  }

  &.info {
    @apply text-textInfoDarker border-bgInfoDarker hover:bg-bgInfoLighter/[.05];
    @apply dark:text-dark-textInfoDarker dark:border-dark-bgInfoDarker hover:dark:bg-dark-bgInfoLighter/[.05];
  }
}

.text {
  &.primary {
    @apply text-textPrimaryDarker hover:bg-bgPrimaryLight/[.1];
    @apply dark:text-dark-textPrimaryDarker hover:dark:bg-dark-bgPrimaryLight/[.1];
  }

  &.secondary {
    @apply text-textSecondaryDarker hover:bg-bgSecondaryLight/[.1];
    @apply dark:text-dark-textSecondaryDarker hover:dark:bg-dark-bgSecondaryLight/[.1];
  }

  &.success {
    @apply text-textSuccessDarker hover:bg-bgSuccessLight/[.1];
    @apply dark:text-dark-textSuccessDarker hover:dark:bg-dark-bgSuccessLight/[.1];
  }

  &.danger {
    @apply text-textDangerDarker hover:bg-bgDangerLight/[.1];
    @apply dark:text-dark-textDangerDarker hover:dark:bg-dark-bgDangerLight/[.1];

  }

  &.warning {
    @apply text-textWarningDarker hover:bg-bgWarningLight/[.1];
    @apply dark:text-dark-textWarningDarker hover:dark:bg-dark-bgWarningLight/[.1];
  }

  &.info {
    @apply text-textInfoDarker hover:bg-bgInfoLight/[.1];
    @apply dark:text-dark-textInfoDarker hover:dark:bg-dark-bgInfoLight/[.1];
  }
}

