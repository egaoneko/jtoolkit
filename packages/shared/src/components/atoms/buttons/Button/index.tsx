import { ButtonHTMLAttributes, MouseEvent, FC, useCallback, useRef, ReactNode } from 'react';
import clsx from 'clsx';

import { Color } from '../../../../enums/color';
import { ButtonGroupDirection, ButtonVariant } from '../../../../enums/button';
import { Size } from '../../../../enums/size';
import Ripple from '../../effect/Ripple';

import styles from './index.module.pcss';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: Size;
  color?: Color;
  disabled?: boolean;
  rounded?: boolean;
  popped?: boolean;
  gradient?: boolean;
  relief?: boolean;
  ripple?: boolean;
  rippleCenter?: boolean;
  group?: boolean;
  groupDirection?: ButtonGroupDirection;
  startIcon?: ReactNode;
  endIcon?: ReactNode;
  className?: string;
}

const Button: FC<ButtonProps> = ({
  variant = ButtonVariant.CONTAINED,
  size = Size.MEDIUM,
  color = Color.PRIMARY,
  disabled = false,
  rounded = false,
  popped = false,
  gradient = false,
  relief = false,
  ripple = true,
  rippleCenter = false,
  group = false,
  groupDirection = ButtonGroupDirection.ROW,
  startIcon,
  endIcon,
  onClick,
  className,
  children,
  ...props
}) => {
  const buttonRef = useRef<HTMLButtonElement>(null);

  const handleOnClick = useCallback(
    (event: MouseEvent<HTMLButtonElement>) => {
      onClick?.(event);
    },
    [onClick],
  );

  return (
    <button
      ref={buttonRef}
      className={clsx(
        styles.button,
        styles[variant],
        styles[size],
        styles[color],
        {
          [styles.rounded]: rounded,
          [styles.gradient]: gradient,
          [styles.relief]: relief,
          [styles.group]: group,
          [styles[groupDirection]]: group,
          ['transform hover:scale-110']: !group && (popped || gradient),
        },
        className,
      )}
      disabled={disabled}
      onClick={handleOnClick}
      {...props}
    >
      {startIcon && <span>{startIcon}</span>}
      {children}
      {endIcon && <span>{endIcon}</span>}
      {ripple && <Ripple parentRef={buttonRef} center={rippleCenter} />}
    </button>
  );
};

export default Button;
