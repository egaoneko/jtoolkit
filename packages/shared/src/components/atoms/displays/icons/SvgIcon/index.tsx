import { forwardRef, SVGAttributes } from 'react';
import clsx from 'clsx';

import { Size } from 'src/enums/size';

import styles from './index.module.css';

export interface SvgIconProps extends SVGAttributes<SVGSVGElement> {
  size?: Size;
  stroked?: boolean;
  filled?: boolean;
  viewBox?: string;
  className?: string;
}

const SvgIcon = forwardRef<SVGSVGElement, SvgIconProps>(function SvgIcon(
  { size = Size.MEDIUM, stroked = false, filled = true, viewBox = '0 0 24 24', className, children, ...rest },
  ref,
) {
  return (
    <svg
      ref={ref}
      className={clsx(styles.svg, styles[size], { [styles.stroked]: stroked, [styles.filled]: filled }, className)}
      width="1em"
      height="1em"
      viewBox={viewBox}
      {...rest}
    >
      {children}
    </svg>
  );
});

export default SvgIcon;
