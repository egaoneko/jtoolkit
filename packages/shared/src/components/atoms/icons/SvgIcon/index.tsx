import { FC, SVGAttributes } from 'react';
import clsx from 'clsx';

import { Size } from '../../../../enums/size';

import styles from './index.module.pcss';

export interface SvgIconProps extends SVGAttributes<SVGElement> {
  size?: Size;
  stroked?: boolean;
  filled?: boolean;
  viewBox?: string;
  className?: string;
}

const SvgIcon: FC<SvgIconProps> = ({
  size = Size.MEDIUM,
  stroked = false,
  filled = true,
  viewBox = '0 0 24 24',
  className,
  children,
  ...rest
}) => {
  return (
    <svg
      {...rest}
      className={clsx(styles.svg, styles[size], { [styles.stroked]: stroked }, { [styles.filled]: filled }, className)}
      width="1em"
      height="1em"
      viewBox={viewBox}
    >
      {children}
    </svg>
  );
};

export default SvgIcon;
