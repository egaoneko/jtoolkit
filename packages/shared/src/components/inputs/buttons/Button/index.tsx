import { ButtonHTMLAttributes, useRef, ReactNode, forwardRef } from 'react';
import clsx from 'clsx';
import { useForkRef } from '@toolkit/util';

import { Color } from 'src/enums/color';
import { ButtonGroupDirection, ButtonVariant } from 'src/enums/button';
import { Size } from 'src/enums/size';
import Ripple from 'src/components/bases/effects/Ripple';
import styles from 'src/components/inputs/buttons/Button/index.module.css';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: Size;
  color?: Color;
  disabled?: boolean;
  rounded?: boolean;
  popped?: boolean;
  gradient?: boolean;
  relief?: boolean;
  ripple?: boolean;
  rippleCenter?: boolean;
  group?: boolean;
  groupDirection?: ButtonGroupDirection;
  startIcon?: ReactNode;
  endIcon?: ReactNode;
  className?: string;
}

const Button = forwardRef<HTMLButtonElement, ButtonProps>(function Button(
  {
    variant = ButtonVariant.CONTAINED,
    size = Size.MEDIUM,
    color = Color.PRIMARY,
    disabled = false,
    rounded = false,
    popped = false,
    gradient = false,
    relief = false,
    ripple = true,
    rippleCenter = false,
    group = false,
    groupDirection = ButtonGroupDirection.ROW,
    startIcon,
    endIcon,
    className,
    children,
    ...props
  },
  ref,
) {
  const buttonRef = useRef<HTMLButtonElement>(null);
  const handleRef = useForkRef(buttonRef, ref);

  return (
    <button
      ref={handleRef}
      className={clsx(
        styles.button,
        styles[variant],
        styles[size],
        styles[color],
        {
          [styles.rounded]: rounded,
          [styles.gradient]: gradient,
          [styles.relief]: relief,
          [styles.popped]: !group && popped,
          [styles.group]: group,
          [styles[groupDirection]]: group,
        },
        className,
      )}
      disabled={disabled}
      {...props}
    >
      {startIcon && <span>{startIcon}</span>}
      {children}
      {endIcon && <span>{endIcon}</span>}
      {ripple && <Ripple parentRef={buttonRef} center={rippleCenter} />}
    </button>
  );
});

export default Button;
