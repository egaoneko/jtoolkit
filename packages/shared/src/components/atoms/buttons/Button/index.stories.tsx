import { Story } from '@storybook/react';

import { Color } from '../../../../enums/color';
import { ButtonVariant } from '../../../../enums/button';
import { Size } from '../../../../enums/size';

import Button, { ButtonProps } from './index';

export default {
  title: 'atoms/buttons/Button',
  component: Button,
};

const StandardTemplate: Story<ButtonProps> = args => (
  <div>
    <Button {...args} variant={ButtonVariant.CONTAINED}>
      Contained
    </Button>
    <Button {...args} variant={ButtonVariant.OUTLINED}>
      Outlined
    </Button>
    <Button {...args} variant={ButtonVariant.TEXT}>
      Text
    </Button>
  </div>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  size: Size.DEFAULT,
  color: Color.PRIMARY,
  disabled: false,
  rounded: false,
  popped: false,
  gradient: false,
  relief: false,
  ripple: true,
};
