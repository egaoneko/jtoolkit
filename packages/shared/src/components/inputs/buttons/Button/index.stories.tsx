import { Story } from '@storybook/react';

import { Color } from 'src/enums/color';
import { ButtonVariant } from 'src/enums/button';
import { Size } from 'src/enums/size';
import SvgIcon from 'src/components/displays/icons/SvgIcon';
import styles from 'src/styles/storybook.css';
import Button, { ButtonProps } from 'src/components/inputs/buttons/Button/index';

export default {
  title: 'atoms/inputs/buttons/Button',
  component: Button,
};

const StandardTemplate: Story<ButtonProps> = args => (
  <div className={styles.row}>
    <Button {...args} variant={ButtonVariant.CONTAINED}>
      Contained
    </Button>
    <Button {...args} variant={ButtonVariant.OUTLINED}>
      Outlined
    </Button>
    <Button {...args} variant={ButtonVariant.TEXT}>
      Text
    </Button>
  </div>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  size: Size.MEDIUM,
  color: Color.PRIMARY,
  disabled: false,
  rounded: false,
  popped: false,
  gradient: false,
  relief: false,
  ripple: true,
  rippleCenter: false,
};

const IconTemplate: Story<ButtonProps> = args => {
  const icon = (
    <SvgIcon size={args.size} color={args.color}>
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
    </SvgIcon>
  );

  return (
    <div className={styles.row}>
      <Button {...args} startIcon={icon}>
        Start
      </Button>
      <Button {...args} endIcon={icon}>
        End
      </Button>
    </div>
  );
};

export const Icon = IconTemplate.bind({});
Icon.args = {
  size: Size.MEDIUM,
  color: Color.PRIMARY,
  variant: ButtonVariant.CONTAINED,
};
