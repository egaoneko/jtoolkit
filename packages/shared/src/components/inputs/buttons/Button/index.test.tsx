import renderer, { ReactTestInstance } from 'react-test-renderer';

import Button from 'src/components/inputs/buttons/Button/index';
import { ButtonGroupDirection, ButtonVariant, Color, Size } from 'src/enums';
import SvgIcon from 'src/components/displays/icons/SvgIcon';

describe('Button Snapshots', () => {
  const icon = (
    <SvgIcon>
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
    </SvgIcon>
  );

  test('renders Button with default options', async () => {
    const tree = renderer.create(
      <Button>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).type).toBe('div');
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0]).toBe('Test');
    expect(tree.toTree()?.props.className).toContain('contained');
    expect(tree.toTree()?.props.className).toContain('medium');
    expect(tree.toTree()?.props.className).toContain('primary');
  });

  test('renders Button with class', async () => {
    const tree = renderer.create(
      <Button className="test">
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('test');
  });

  test('renders Button with variant', async () => {
    const tree = renderer.create(
      <Button variant={ButtonVariant.TEXT}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain(ButtonVariant.TEXT);
  });

  test('renders Button with size', async () => {
    const tree = renderer.create(
      <Button size={Size.SMALL}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain(Size.SMALL);
  });

  test('renders Button with color', async () => {
    const tree = renderer.create(
      <Button color={Color.SECONDARY}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain(Color.SECONDARY);
  });

  test('renders Button with rounded', async () => {
    const tree = renderer.create(
      <Button rounded>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('rounded');
  });

  test('renders Button with gradient', async () => {
    const tree = renderer.create(
      <Button gradient>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('gradient');
  });

  test('renders Button with relief', async () => {
    const tree = renderer.create(
      <Button relief>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('relief');
  });

  test('renders Button with popped', async () => {
    const tree = renderer.create(
      <Button popped>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('popped');
  });

  test('renders Button popped with group', async () => {
    const tree = renderer.create(
      <Button popped group>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toTree()?.props.className).not.toContain('popped');
  });

  test('renders Button with startIcon', async () => {
    const tree = renderer.create(
      <Button startIcon={icon}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).type).toBe('span');
    expect(((tree.root.children[0] as ReactTestInstance).children[1] as ReactTestInstance).type).toBe('div');
  });

  test('renders Button with endIcon', async () => {
    const tree = renderer.create(
      <Button endIcon={icon}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).type).toBe('div');
    expect(((tree.root.children[0] as ReactTestInstance).children[1] as ReactTestInstance).type).toBe('span');
  });

  test('renders Button with icons', async () => {
    const tree = renderer.create(
      <Button startIcon={icon} endIcon={icon}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).type).toBe('span');
    expect(((tree.root.children[0] as ReactTestInstance).children[1] as ReactTestInstance).type).toBe('div');
    expect(((tree.root.children[0] as ReactTestInstance).children[2] as ReactTestInstance).type).toBe('span');
  });

  test('renders Button with group', async () => {
    const tree = renderer.create(
      <Button group>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('group');
    expect(tree.toTree()?.props.className).toContain('row');
  });

  test('renders Button with group column', async () => {
    const tree = renderer.create(
      <Button group groupDirection={ButtonGroupDirection.COLUMN}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('group');
    expect(tree.toTree()?.props.className).toContain('column');
  });

  test('renders Button without ripple', async () => {
    const tree = renderer.create(
      <Button ripple={false}>
        <div>Test</div>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect((tree.root.children[0] as ReactTestInstance).children.length).toBe(1);
  });
});
