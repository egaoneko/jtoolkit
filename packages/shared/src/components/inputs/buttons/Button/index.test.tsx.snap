// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots renders Button with class 1`] = `
<button
  className="button contained medium primary test"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with color 1`] = `
<button
  className="button contained medium secondary"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with default options 1`] = `
<button
  className="button contained medium primary"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with endIcon 1`] = `
<button
  className="button contained medium primary"
  disabled={false}
>
  <div>
    Test
  </div>
  <span>
    <svg
      className="svg medium filled"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </span>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with gradient 1`] = `
<button
  className="button contained medium primary gradient"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with group 1`] = `
<button
  className="button contained medium primary group row"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with group column 1`] = `
<button
  className="button contained medium primary group column"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with icons 1`] = `
<button
  className="button contained medium primary"
  disabled={false}
>
  <span>
    <svg
      className="svg medium filled"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </span>
  <div>
    Test
  </div>
  <span>
    <svg
      className="svg medium filled"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </span>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with popped 1`] = `
<button
  className="button contained medium primary popped"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with relief 1`] = `
<button
  className="button contained medium primary relief"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with rounded 1`] = `
<button
  className="button contained medium primary rounded"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with size 1`] = `
<button
  className="button contained small primary"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with startIcon 1`] = `
<button
  className="button contained medium primary"
  disabled={false}
>
  <span>
    <svg
      className="svg medium filled"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </span>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button with variant 1`] = `
<button
  className="button text medium primary"
  disabled={false}
>
  <div>
    Test
  </div>
  <span
    className="ripple"
  />
</button>
`;

exports[`Button Snapshots renders Button without ripple 1`] = `
<button
  className="button contained medium primary"
  disabled={false}
>
  <div>
    Test
  </div>
</button>
`;
